### PR TITLE
feat: adding a CB trigger and periodic job for purge untagged CFT images

### PR DESCRIPTION
**Pull request by g-awmalik**

### CODE REVIEW

File: infra/terraform/test-org/image-cleanup/cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    dir: "infra/terraform/test-org/image-cleanup"
+    id: "purge-dev-tools"
+    entrypoint: "bash"
+    args:
+      [
+        "./cft-image-cleanup.sh",
+        "$_REGISTRY_URL/$_DOCKER_IMAGE_DEVELOPER_TOOLS",
+        "DELETE",
+      ]
+  - name: "gcr.io/cloud-builders/gcloud"
```
Comment (apeabody) [inline infra/terraform/test-org/image-cleanup/cloudbuild.yaml:26]: Not a big deal with a scheduled job, but you could add `waitFor: ['-'] ` to each to all steps start immediately.

File: infra/terraform/test-org/image-cleanup/cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+steps:
+  - name: "gcr.io/cloud-builders/gcloud"
+    dir: "infra/terraform/test-org/image-cleanup"
+    id: "purge-dev-tools"
+    entrypoint: "bash"
+    args:
+      [
+        "./cft-image-cleanup.sh",
+        "$_REGISTRY_URL/$_DOCKER_IMAGE_DEVELOPER_TOOLS",
+        "DELETE",
```
Comment (apeabody) [inline infra/terraform/test-org/image-cleanup/cloudbuild.yaml:24]: Might initially want to run these in dry-run for a day just to make sure everything is working as expected?

```suggestion
```

**Review by apeabody — APPROVED** [review]
Thanks @g-awmalik!  A few suggestions

File: infra/terraform/test-org/image-cleanup/cloudbuild.yaml (hunk 2 above)
Comment (g-awmalik) [reply infra/terraform/test-org/image-cleanup/cloudbuild.yaml:24]: I did run them in dry run pointing to my branch. Works fine.

File: infra/terraform/test-org/image-cleanup/cloudbuild.yaml (hunk 1 above)
Comment (g-awmalik) [reply infra/terraform/test-org/image-cleanup/cloudbuild.yaml:27]: done

**Pull request merged**